plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -324,20 +360,42 @@ open class KtlintPlugin : Plugin<Project> {
             description = "Runs ktlint on all kotlin sources in this project."
         }
 
+    private fun Project.getAndroidVariantMetaKtlintCheckTask(
+        variantName: String,
+        multiplatformTargetName: String? = null
+    ): Task {
+        val taskName = "ktlint${variantName.capitalize()}${multiplatformTargetName?.capitalize() ?: ""}Check"
+        return tasks.findByName(taskName) ?: task(taskName).apply {
```

Comment:
Instead can you use `tasks.maybeCreate()` that conditionally creates a task and returns the already available task if so.

Reply:
Unfortunately, all of these options are eagerly evaluated. I'd love if there was a way to lazily instantiate this task.

Reply:
I'm not sure actually. findByName is also not lazy. `maybeCreate` looks like a drop-in replacement of what you do here. 

At the end it is not crucial. Either is fine.

Reply:
I think, this can addressed in next PR with task configuration avoidance. For now, I changed it to `maybeCreate()`